spyder/plugins/editor/lsp/manager.py

```diff
@@ -164,6 +167,9 @@ def start_client(self, language):
                 if self.main and self.main.editor:
                     language_client['instance'].sig_initialize.connect(
                         self.main.editor.register_lsp_server_settings)
+                elif running_under_pytest():
```

Comment:
This was created for the tests, but I see no "harm" in leaving it always connected?

Reply:
I think it's more elegant to use a Mock for the editor plugin and main window in `lsp_manager`. That would avoid this.

Reply:
Let me address this and I'll let you know when it's ready for another review.

Reply:
I address this in my last commit. Please review again.